src/kOS/Module/kOSProcessor.cs

```diff
@@ -528,8 +528,15 @@ public void SetMode(ProcessorModes newProcessorMode)
                 switch (newProcessorMode)
                 {
                     case ProcessorModes.READY:
-                        if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)
-                            shared.Cpu.Boot();
+                        if (Config.Instance.StartOnArchive)
```

Comment:
The removal of :
```
if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)
```
Is very wrong.  If one part of the code calls SetMode(ProcessorModes.READY) and then another part of the code calls it again unnecessarily, you don't want that to cause a reboot.  It should only reboot when it wasn't already in the READY setting to begin with.

You could change it to check for ```if ((ProcessorMode != ProcessorModes.READY)...etc``` instead, so it really does work into the future if more states are added, but I don't like the idea that setting it to READY when it was already READY causes a reboot.